File: 1주차/PGS_폰켓몬/seoheun.py
def solution(nums):
    num = len(nums)/2
    num_type = len(list(set(nums)))
    
    if num <= num_type:
        # 만약 종류가 고를 수 있는 개수와 같거나 클 경우 -> num개
        return num
    else:
        # 종류가 고를 수 있는 개수보다 작을 경우 -> num_type개
        return num_type

File: 4주차/PGS_소수 찾기/seoheun.js
// 검색 참고

function solution(numbers) {
    var answer = [];

    let num_arr = numbers.split("");

    const isPrime = (num) => {
        if (num === 1 || num === 0) return false;
        for (let i = 2; i * i <= num; i++) {
            if (num % i === 0) return false;
        }
        // 소수일 경우
        return true;
    };

    const DFS = (arr, str) => {
        if (arr.length >= 1) {
            for (let i = 0; i < arr.length; i++) {
                const temp = Math.abs(str + arr[i]);
                const nums = [...arr];
                nums.splice(i, 1);
                if (!answer.includes(temp) && isPrime(temp)) {
                    answer.push(temp);
                }
                DFS(nums, temp);
            }
        }
    };

    DFS(num_arr, "");
    return answer.length;
}
